test/e2e/framework/service_util.go

```diff
@@ -756,6 +819,106 @@ func (j *ServiceTestJig) waitForPodsReady(namespace string, pods []string) error
 	return nil
 }
 
+func testReachabilityOverServiceName(serviceName string, sp v1.ServicePort, execPod *v1.Pod) {
+	testEndpointReachability(serviceName, sp.Port, sp.Protocol, execPod)
+}
+
+func testReachabilityOverClusterIP(clusterIP string, sp v1.ServicePort, execPod *v1.Pod) {
+	// If .spec.clusterIP is set to "" or "None" for service, ClusterIP is not created, so reachability can not be tested over clusterIP:servicePort
+	isClusterIPV46, err := regexp.MatchString(RegexIPv4+"||"+RegexIPv6, clusterIP)
+	ExpectNoError(err, "Unable to parse ClusterIP: %s", clusterIP)
+	if isClusterIPV46 {
+		testEndpointReachability(clusterIP, sp.Port, sp.Protocol, execPod)
+	}
+}
+func testReachabilityOverNodePorts(nodes *v1.NodeList, sp v1.ServicePort, pod *v1.Pod) {
+	internalAddrs := e2enode.CollectAddresses(nodes, v1.NodeInternalIP)
+	externalAddrs := e2enode.CollectAddresses(nodes, v1.NodeExternalIP)
+	for _, internalAddr := range internalAddrs {
+		testEndpointReachability(internalAddr, sp.NodePort, sp.Protocol, pod)
+	}
+	for _, externalAddr := range externalAddrs {
+		testEndpointReachability(externalAddr, sp.NodePort, sp.Protocol, pod)
+	}
+}
+
+// testEndpointReachability tests reachability to endpoints (i.e. IP, ServiceName) and ports. Test request is initiated from specified execPod.
+func testEndpointReachability(endpoint string, port int32, protocol v1.Protocol, execPod *v1.Pod) {
+	ep := net.JoinHostPort(endpoint, strconv.Itoa(int(port)))
+	cmd := ""
+	switch protocol {
+	case v1.ProtocolTCP:
+		cmd = fmt.Sprintf("nc -z -t -w 2 %s %v", endpoint, port)
+	case v1.ProtocolUDP:
+		cmd = fmt.Sprintf("nc -z -u -w 2 %s %v", endpoint, port)
+	}
```

Comment:
This still fails silently for the `ProtocolSCTP` case right?

Reply:
`ProtocolSCTP` based service reachability is not handled at this moment and is in TODO. Definitely e2e would not fail and would bypass service reachability check. Currently we do not have an image exposing SCTP service. Following would be in TODO -
1. Add `SCTP` service support in `agnhost`.
2. Add `SCTP` service reachability test command here.

Would it be blocking this PR to get merged? or TODO can be incorporated asap `SCTP` service support is added into `agnhost` image.

Reply:
Can we add a `default` fallback on the switch statement and call Failf? The main concern is someone passing SCTP protocol in their new tests and thinking it passed reachibility tests.

Reply:
Thanks for suggestion. Added default: case to fail incase unsupported (At this moment) protocol is observed because SCTP is supposed to be tested for enduser as it's valid for them but test suite is not capable of testing it.

TODO must be addressed at earliest.